packages/admin/cms-admin/src/generator/future/generateGrid/combinationColumn.ts

```diff
@@ -14,15 +14,11 @@ type StaticText = {
     text: string;
 };
 
-// type NumberField<FieldName extends string> = AbstractField<FieldName> & {
-//     type: "number";
-//     decimals?: number;
-// };
-
-// type CurrencyField<FieldName extends string> = AbstractField<FieldName> & {
-//     type: "currency";
-//     currency: string;
-// };
+type NumberField<FieldName extends string> = AbstractField<FieldName> & {
+    type: "number";
+    decimals?: number;
+    currency?: string;
+};
```

Comment:
we should also support other units (eg. weight in kg), can't this be generic instead of currency specific?

Reply:
It can't really be generic like that, as `currency` doesn't just add something to the beginning/end of the value but sets `style="currency"` to `<FormattedNumber />`.

We could expose more, maybe all props of `FormattedNumber`, then we could support all of the following things for example:
 
| JSX                                                                   | Output     |
| --------------------------------------------------------------------- | ---------- |
| `<FormattedNumber value={123.456} style="currency" currency="EUR" />` | €123.46    |
| `<FormattedNumber value={123.456} style="unit" unit="kilobyte" />`    | 123.456 kB |
| `<FormattedNumber value={123.456} style="unit" unit="kilogram" />`    | 123.456 kg |
| `<FormattedNumber value={123.456} style="unit" unit="celsius" />`     | 123.456°C  |
| `<FormattedNumber value={9 / 16} style="percent" />`                  | 56%        |

What do you think of this?
Should we maybe add `prefix` and `postfix` keys to the config to allow setting units? Maybe in addition to exposing the props 🤔

Reply:
Exposing those props seems a good idea.